packages/nimbus/src/components/text-input/text-input.mdx

```diff
@@ -13,106 +13,180 @@ tags:
+    * Display error messages close to the input field.
+* **Placeholder text:**
+    * Avoid relying solely on placeholder text, as it disappears when the user starts typing.
+    * Make use of clear labels and helper text if clarifying information is needed.
 
-For showing error states in forms.
+### Placeholder text
+
+Good placeholder text informs the user what to expect and can easily understand the difference between placeholder and pre-filled content.
+
+> [!TIP]\
+> When to use
+
+- **Provide hints:** Use placeholder text to give users a brief hint or example of the expected input format.
+- **Keep it concise:** Placeholder text should be short and to the point. Avoid long sentences or paragraphs.
+- **Use a subtle visual style:** Visually differentiate placeholder text from actual input text by using the placeholder text styling. This helps users understand that it's a suggestion, not a pre-filled value.  
+
+> [!CAUTION]\
+> When not to use
+
+- **Don't use as labels:** Placeholder text should never replace proper labels. Labels provide persistent context, while placeholder text disappears when the user starts typing.  
+- **Don't include critical information:** Avoid putting essential instructions or information solely in the placeholder, as it will be hidden once the user interacts with the field.  
+- **Don't overuse:** Too much placeholder text can clutter the interface and make it harder for users to focus.
+
+> [!TIP]\
+> **Do**
+>
+> - Use the correct tone for placeholder text.
+> - Give the user a hint on what is expected, in a concise way. 
+> - If this information is important to always keep visible, use helper text instead.
 
 ```jsx-live
-const App = () => {
-
-  const variants = ['solid','ghost',];
-
-  return (
-    <Stack direction="horizontal">
-      {variants.map(variant => (
-        <TextInput key={variant} variant={variant} placeholder={variant} isInvalid/>
-      ))}
-    </Stack>
-  )
-}
+const App = () => <TextInput tone="critical" placeholder="MM/DD/YYYY"/>
 ```
 
+> [!CAUTION]\
+> **Don't**
+>
+> - This placeholder text looks like it has been prefilled for the user which is more likely to confuse the user. 
+
+```jsx-live
+const App = () => <TextInput tone="critical" placeholder="01/21/1970"/>
+```
+
+### Size
+
+Allow for enough width for the user to read and add in their information. Avoid truncation if possible especially if the user will be referring back or needing to make sure there isn’t inaccurate information.  
+
+> [!TIP]\
+> When to use
+> 
+> - There is enough space that the user can easily consume and correct this information if needed.
+
+```jsx-live
+const App = () => <TextInput placeholder="01/21/1970" value="805 St Cloud Road"/>
+```
+
+> [!CAUTION]\
+> **Don't**
+>
+> - The user’s information is too long for this field to be able to read the details completely.
+
+```jsx-live
+const App = () => <TextInput value="805 St Cloud Road, Bel Air, Los Angeles, CA" />
```

Comment:
Should the text input value be truncated with ellipses? 
![image](https://github.com/user-attachments/assets/8d4c90e9-2269-4688-b938-1638014196d4)

Reply:
I think maybe that is the purpose of this example. If the users details are too long, maybe then another kind of input field should be used or the user information should be broken down. Also, since our text-input is almost close to the native one, we really did not modify much, I think it is safer to keep it this way than to try to modify the displayed value. 
<img width="753" alt="image" src="https://github.com/user-attachments/assets/1701c5dd-83c5-4b1b-97c4-2da10faa7347" />